### PR TITLE
[2784] Add more monitored functions

### LINKED ISSUE #2784: Diagrams can be slow to open
In some use cases, visible in projects like Eclipse Papyrus Web or Eclipse SysOn, some diagrams can be slow to display. 

While part of the issue may come from our new integration with ReactFlow, the fact that the diagram panel can take a couple seconds to be displayed makes me believe that something may be going on on the backend side of the application.

The first goal of this issue is to ensure that we have the proper monitoring capabilities, based on on micrometer on the backend, in order to evaluate the time spent on the various steps of the following use cases:

- with a simple syson project, when I open an empty diagram, then I want the diagram to be loaded quickly
- with a simple syson project, when I switch from one diagram to another (after more than six seconds on the current diagram), then I want the diagram to be loaded quickly

The following steps should be monitored to have some timing data:

- loading of the editing context event processor
- loading of the diagram event processor
- refresh of the representation
- sending the diagram refresh event payload to the frontend
- time to process each input (to detect that an input is being processed while another one has been sent)
- rendered of the diagram on the frontend (using something like console.log on the frontend)

We should be able to retrieve monitoring data from an instance of syson running locally using prometheus and grafana and after that monitor how things are running on the real demo instance on AWS. Once we will have those data, we will see how to proceed to fix the issues detected.

### PR DESCRIPTION
**Pull request by mcharfadi**
Bug: https://github.com/eclipse-sirius/sirius-web/issues/2784

# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [ ] New features
- [ ] Documentation
- [ ] Cleanup
- [x] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: packages/core/backend/sirius-components-collaborative/src/main/java/org/eclipse/sirius/components/collaborative/editingcontext/EditingContextEventProcessor.java [hunk 1]
```diff
@@ -212,6 +218,7 @@ public String getEditingContextId() {
 
     @Override
     public Mono<IPayload> handle(IInput input) {
+        Timer doHandleTimer = Timer.builder(Monitoring.EDITING_CONTEXT_EVENT_PROCESSOR_LOADING).tag("inputId", input.id().toString()).register(meterRegistry);
```
Comment (mcharfadi) [inline packages/core/backend/sirius-components-collaborative/src/main/java/org/eclipse/sirius/components/collaborative/editingcontext/EditingContextEventProcessor.java]: not sure about keeping the id as a tag here

File: packages/core/backend/sirius-components-collaborative/src/main/java/org/eclipse/sirius/components/collaborative/api/Monitoring.java [hunk 2]
```diff
@@ -23,6 +23,10 @@ public final class Monitoring {
 
     public static final String REPRESENTATION_EVENT_PROCESSOR_REFRESH = "siriusweb_representationeventprocessor_refresh";
 
+    public static final String EDITING_CONTEXT_EVENT_PROCESSOR_LOADING = "siriusweb_edtingcontexteventprocessor_loading";
+
+    public static final String DIAGRAM_EVENT_PROCESSOR_LOADING = "siriusweb_diagrameventprocessor_loading";
```
Comment (sbegaudeau) [inline packages/core/backend/sirius-components-collaborative/src/main/java/org/eclipse/sirius/components/collaborative/api/Monitoring.java]: Do not reference any specific representation here, no diagram nor any other representation

File: packages/charts/backend/sirius-components-collaborative-charts/src/main/java/org/eclipse/sirius/components/collaborative/charts/HierarchyCreationService.java [hunk 3]
```diff
@@ -61,11 +62,13 @@ public HierarchyCreationService(IRepresentationDescriptionSearchService represen
         // @formatter:on
     }
 
+    @Timed(value = Monitoring.REPRESENTATION_EVENT_PROCESSOR_REFRESH, extraTags = {Monitoring.NAME, "hierarchy"})
```
Comment (sbegaudeau) [inline packages/charts/backend/sirius-components-collaborative-charts/src/main/java/org/eclipse/sirius/components/collaborative/charts/HierarchyCreationService.java]: Can't all those `@Timed` annotation on the creation of the representation be instead put around the creation of the representation event processor factory in `EditingContextEventProcessor#aquireRepresentationEventProcessor` in order to have something generic and not tied to some representation-specific service

File: packages/charts/backend/sirius-components-collaborative-charts/src/main/java/org/eclipse/sirius/components/collaborative/charts/HierarchyCreationService.java [hunk 4]
```diff
@@ -61,11 +62,13 @@ public HierarchyCreationService(IRepresentationDescriptionSearchService represen
...
     public Hierarchy create(String label, Object targetObject, HierarchyDescription hierarchyDescription, IEditingContext editingContext) {
         Hierarchy newHierarchy = this.doRender(label, targetObject, editingContext, hierarchyDescription, Optional.empty());
         return newHierarchy;
     }
 
+    @Timed(value = Monitoring.REPRESENTATION_EVENT_PROCESSOR_REFRESH, extraTags = {Monitoring.NAME, "hierarchy"})
```
Comment (sbegaudeau) [inline packages/charts/backend/sirius-components-collaborative-charts/src/main/java/org/eclipse/sirius/components/collaborative/charts/HierarchyCreationService.java]: Can't all those `@Timed` annotation on the refresh of the representation be instead put around the refresh of the representation event factory in `EditingContextEventProcessor#setupChangeDescriptionSinkConsumer` and `EditingContextEventProcessor#refreshOtherRepresentations(...)` when calling representationEventProcessor.refresh(...)`. I don't want to have to tell all those who will write their own representation to setup things like that.

File: packages/core/backend/sirius-components-collaborative/src/main/java/org/eclipse/sirius/components/collaborative/editingcontext/EditingContextEventProcessor.java [hunk 5]
```diff
@@ -110,7 +113,9 @@ public class EditingContextEventProcessor implements IEditingContextEventProcess
     
     private final List<IInputPostProcessor> inputPostProcessors;
 
-    public EditingContextEventProcessor(EditingContextEventProcessorParameters parameters) {
+    private final MeterRegistry meterRegistry;
+
+    public EditingContextEventProcessor(EditingContextEventProcessorParameters parameters, MeterRegistry meterRegistry) {
```
Comment (sbegaudeau) [inline packages/core/backend/sirius-components-collaborative/src/main/java/org/eclipse/sirius/components/collaborative/editingcontext/EditingContextEventProcessor.java]: Please put the `meterRegistry` in the `parameters`, same thing for the representations event processor if that's needed

File: packages/charts/backend/sirius-components-collaborative-charts/src/main/java/org/eclipse/sirius/components/collaborative/charts/CreateHierarchyEventHandler.java [hunk 6]
```diff
@@ -88,6 +89,7 @@ public boolean canHandle(IEditingContext editingContext, IInput input) {
     }
 
     @Override
+    @Timed(value = Monitoring.EVENT_HANDLER)
```
Comment (sbegaudeau) [inline packages/charts/backend/sirius-components-collaborative-charts/src/main/java/org/eclipse/sirius/components/collaborative/charts/CreateHierarchyEventHandler.java]: Why do we need to modify around 100 files to add those kind of timers if we monitor the time taken by the execution of an input in `EditingContextEventProcessor#handle`? I don't think we will spend that much time iterating on the representation event processors to find the right one and on the event handlers to find the right one too.

**Review by sbegaudeau — CHANGES_REQUESTED** [review]

**Review by sbegaudeau — APPROVED** [review]

**Pull request merged**